deps/3rdparty/libdispatch/src/queue.c

```diff
@@ -2039,7 +2039,7 @@ static void _dispatch_queue_insert_specific(dispatch_queue_specific_t dqs) {
 	TAILQ_FOREACH(var, &dqsl->context_list, specific) {
 		if (var->key == dqs->key) {
 			if(var->destructor) {
-				var->destructor(var->context);
+				dispatch_async_f(_dispatch_get_root_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, false), var->context, var->destructor);
```

Comment:
Should this be default priority, or lower?

Reply:
I don't see why it should be lower. It's not a high priority item, but I can't imagine the user wants to wait longer than they have to.

Reply:
Waiting on a destructor is not likely to be a very common case though :)

Reply:
Since it's their only way of freeing those resources though, we shouldn't be a barrier to getting it done.